website/lib/Waypointy.class.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+class Waypointy
+{
+    // database session opened with DBPConnect();
+    public $dblink;
+    // report current activity to stdout
+    public $verbose;
+
+    //~ waypoint attributes
+    public $waypoint;
+    public $lat;
+    public $lon;
+    public $name;
+    public $owner;
+    public $typ;
+    public $cache_type;
+    public $cache_link;
+    public $kraj;
+    public $country;
+
+    public function __construct($dblink, $verbose)
+    {
+        $this->dblink = $dblink;
+        $this->verbose = $verbose;
+    }
+
+    public function getByWaypoint($waypoint)
+    {
+        if (!isset($waypoint) || $waypoint == '') {
+            throw new Exception('Waypoint expected');
+        }
+        $sqlquery = 'SELECT wp.`waypoint`, wp.`lat`, wp.`lon`, wp.`name`, wp.`owner`, wp.`typ`, wpt.`cache_type`, wp.`link`, wp.`kraj`, wpc.`country`'
+               .' FROM `gk-waypointy` wp'
+               .' JOIN `gk-waypointy-country` wpc ON wp.`kraj` = wpc.`kraj`'
+               .' JOIN `gk-waypointy-type` wpt ON wp.`typ` = wpt.`typ`'
+               .' WHERE wp.`waypoint`=\'%s\''
+               .' LIMIT 1';
+        $query = sprintf($sqlquery, mysqli_real_escape_string($this->dblink, $waypoint));
```

Comment:
It would be even better to use [Sql Prepared Statements](http://php.net/manual/en/mysqli.quickstart.prepared-statements.php).

What about: (adapted from the doc)
```
if (!($stmt = $this->dblink->prepare('SELECT wp.`waypoint`, wp.`lat`, wp.`lon`, wp.`name`, wp.`owner`, wp.`typ`, wpt.`cache_type`, wp.`link`, wp.`kraj`, wpc.`country`'
               .' FROM `gk-waypointy` wp'
               .' JOIN `gk-waypointy-country` wpc ON wp.`kraj` = wpc.`kraj`'
               .' JOIN `gk-waypointy-type` wpt ON wp.`typ` = wpt.`typ`'
               .' WHERE wp.`waypoint`=?'
               .' LIMIT 1'))) {
    throw new Exception('Prepare failed: ('.$this->dblink->errno.') '.$this->dblink->error);
}

if (!$stmt->bind_param("s", $waypoint)) {
    throw new Exception('Binding parameters failed: ('.$stmt->errno.') '.$stmt->error);
}

if (!$stmt->execute()) {
    throw new Exception('Execute failed: ('.$stmt->errno.') '.$stmt->error);
}

if (!$stmt->bind_result($this->waypoint, $this->lat, $this->lon, $this->name, $this->owner, $this->typ, $this->cache_type, $this->cache_link, $this->kraj, $this->country)) {
    throw new Exception('Binding output parameters failed: ('.$stmt->errno.') '.$stmt->error);
}

if ($stmt->fetch()) {
    return true;
}
return false;
```

Reply:
like it, thanks for this sample. (CR to continue tomorrow)